### PR TITLE
docs: improve readme documentation

### PR DESCRIPTION
**Pull request by maeertin**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,145 @@
...
+```
+
+### src/pages & src/pages.stories.js
+
+The `src/pages` directory is used by Next.js to configure page routing. [Read more about Next.js pages.](https://nextjs.org/docs/basic-features/pages)
+The `src/pages.stories.js` file is used to configure your pages for Storybook.
+
+Both Next.js & Storybook pages use the same `createRenderBlock` helper found under `src/utils/createRenderBlock.js` to iterate over Block components as described from an API. Using the same setup helps to keep a consistent codebase across both environments. Don't forget to add your newly created Block component to the common exports file!
+
+## One template, two enhancers
```
Comment (alexanderflink) [inline README.md:122]: Maybe change "enhancers" to "entry points"? The index files might not contain any enhancing logic.

Comment (maeertin) [reply README.md:122]: Let's change that!

File: README.md [hunk 2]
```diff
@@ -1,9 +1,145 @@
...
+### src/api
+
+This is a good place to add App-level providers that should be accessible from all corners of the application. Providers added here should be exported in the api directory's common export file(s). Accessing data or helpers from these providers can look like this `import { useI18n } from 'api'`.
+
+```
+src/api
+├── __mock__ (Mock data used for Storybook)
+├── CheckoutContext (Checkout-API data & helpers)
+├── I18nContext (Internalization data & helpers)
+├── SettingsContext (CMS data)
```
Comment (alexanderflink) [inline README.md:71]: Maybe "SettingsContext (global settings and CMS data)" ?

Comment (maeertin) [reply README.md:71]: 👍

**Pull request merged**